drivers/clock_control/clock_control_mcux_syscon.c

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2020, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_lpc_syscon
+#include <errno.h>
+#include <soc.h>
+#include <drivers/clock_control.h>
+#include <dt-bindings/clock/mcux_lpc_clock.h>
+#include <fsl_clock.h>
+
+#define LOG_LEVEL CONFIG_CLOCK_CONTROL_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(clock_control);
+
+static int mcux_lpc_clock_control_on(struct device *dev,
```

Comment:
Looks good! I think the function names should be updated to match the new driver name (syscon vs. clock_control).